scripts/build-local.py

```diff
@@ -0,0 +1,165 @@
+        config = yaml.safe_load(file)
+
+    current_jobs = config['jobs']
+
+    for job in config['jobs']:
+        name = job['name']
+        if args.jobs and name not in args.jobs:
+            # This can be noisy under normal circumstances so only show it if
+            # debugging is enabled.
+            if args.debug:
+                jobs_join = "', '"
+                print(
+                    f"D: {Path(file_str).name}: Job '{name}' was not in requested jobs ('{jobs_join.join(args.jobs)}'), skipping..."
+                )
+            continue
+
+        builds = job['builds']
+        if name in jobs:
+            jobs[name] += builds
+        else:
+            jobs[name] = copy.deepcopy(builds)
+
+build_dir = Path(args.build_dir).resolve()
+output_dir = Path(args.output_dir).resolve()
+for name, builds in jobs.items():
+    print(f"I: Executing job '{name}' in {tree}")
+    for build in builds:
+        kconfig = build['kconfig']
+        target_arch = build['target_arch']
+        toolchain = build['toolchain']
+
+        cfg_str = '+'.join(kconfig) if isinstance(kconfig, list) else kconfig
+        print(f"I: Building {target_arch} {cfg_str} ({toolchain})... ",
+              end='\n' if args.verbose else '',
+              flush=True)
+
+        if build_dir.exists():
+            shutil.rmtree(build_dir)
+        build_dir.mkdir(exist_ok=True, parents=True)
+
+        # Replace the URL in the configuration string with a simple name, so
+        # that it can be used in a path.
+        if (match := re.search(r'(https://[^\+]+)', cfg_str)):
+            url = match.groups()[0]
+            if 'alpine' in url:
+                distro = 'alpine'
+            elif 'archlinux' in url:
+                distro = 'archlinux'
+            elif 'fedora' in url:
+                distro = 'fedora'
+            elif 'openSUSE' in url:
+                distro = 'opensuse'
+            cfg_str = cfg_str.replace(url, distro)
+        specific_output_dir = Path(output_dir, toolchain, target_arch, cfg_str)
+        specific_output_dir.mkdir(exist_ok=True, parents=True)
+
+        # If kconfig is a list, we need to split it into 'kconfig' and
+        # 'kconfig_add', as that is what tuxmake expects (I assume tuxsuite
+        # does this internally).
+        if isinstance(kconfig, list):
+            build['kconfig_add'] = build['kconfig'][1:]
+            build['kconfig'] = build['kconfig'][0]
+
+        # Perform the build and report the result
+        result = tuxmake.build.build(build_dir=build_dir,
+                                     output_dir=specific_output_dir,
+                                     quiet=(not args.verbose),
+                                     runtime=runtime,
+                                     tree=tree,
+                                     **build)
+        passed = True
+        for info in result.status.values():
+            passed &= info.passed
+
+        if passed:
```

Comment:
I bet you could use `all` here.

```python
if all(info.passed for info in result.status.values):
```

Reply:
Thanks, that looks much better! 4cb798b007df22ce470547e58824029f1e73cacf